forte/utils/utils.py

```diff
@@ -127,6 +130,57 @@ def get_class(full_class_name: str, module_paths: Optional[List[str]] = None):
     return class_
 
 
+@lru_cache()
+def cached_locate(name_to_locate_class):
+    return locate(name_to_locate_class)
+
+
+def get_class(full_class_name: str, module_paths: Optional[List[str]] = None):
+    r"""Returns the class based on class name, with cache to improve speed.
+
+    Args:
+        class_name (str): Name or full path to the class.
+        module_paths (list): Paths to candidate modules to search for the
+            class. This is used if the class cannot be located solely based on
+            ``class_name``. The first module in the list that contains the class
+            is used.
+
+    Returns:
+        The target class.
+
+    """
+    class_ = cached_locate(full_class_name)
+    if (class_ is None) and (module_paths is not None):
+        for module_path in module_paths:
+            class_ = cached_locate(".".join([module_path, full_class_name]))
+            if class_ is not None:
+                break
+
+    # Try to find classes that are dynamically loaded, class_ will still be None if failed.
+    if class_ is None:
+        try:
+            module_name, class_name = full_class_name.rsplit(".", 1)
+            try:
+                class_ = getattr(sys.modules[module_name], class_name)
+            except (AttributeError, KeyError):
+                # ignore when cannot find the module in sys.modules or cannot find the class
+                # in the module.
+                pass
+        except ValueError:
+            # ignoring when the full class name doesn't have multiple parts.
+            pass
+
+    if class_ is None:
+        if module_paths:
+            raise ValueError(
+                f"Class not found in {module_paths}: {full_class_name}"
+            )
+        else:
+            raise ValueError(f"Class not found in {full_class_name}")
+
+    return class_
```

Comment:
I notice both methods have too many lines duplicated. The lines that are different are basically "locate" or "cached_locate". We should not write it this way. Consider merging the duplicated code as much as possible.

Reply:
This is now fixed: Refactored the code to combine the 2 methods into one "get_class" with "cached_lookup" optional parameter.